lib/breakers/statsd_plugin.rb

```diff
@@ -46,5 +31,30 @@ def send_metric(status, service, request_env, response_env)
         StatsD.measure(metric_base + 'time', response_env[:duration], tags: tags)
       end
     end
+
+    private
+
+    def filtered_endpoint_tag(path)
+      # replace identifiers with 'xxx'
+      # this nasty-looking regex attempts to cover:
+      # * (possibly negative) digit identifiers
+      # * uuid's with or without dashes
+      # * institution id's of form 111A2222 or 11A22222
+      digit = /\-?\d+/
+      contact_id = /\d{10}V\d{6}(%5ENI%5E200M%5EUSVHA)*/
+      uuids = /[a-fA-F0-9]{8}(\-?[a-fA-F0-9]{4}){3}\-?[a-fA-F0-9]{12}/
+      institution_ids = /[\dA-Z]{8}/
+      provider_ids = /Providers\(\d{10}\)/
+      v1_user_ids_1 = /[0-9a-f]{22}$/ # e.g. /api/v1/users/bb9c0f499977be68611151
```

Comment:
are we confident that these are the two types of id's or should we just go for something like "everything after `/api/v1/users/`"?

Reply:
Eh, reasonably confident. I [ran this query](http://prometheus-prod.vetsgov-internal:9090/prometheus/graph?g0.range_input=1h&g0.expr=count%20by%20(endpoint)%20(api_external_http_request_duration_seconds%7Bendpoint%3D~%22%2Fapi%2Fv1%2Fusers%2F.%2B%22%7D)&g0.tab=1), then copy+pasted the output to my text editor, cleaned up the output and "eyeballed" it. At the worst, I've missed some and we'll see them in grafana.

I tried following how other regex's were written in this file. OTOH, I do like how blunt / least-surprise just filtering everything after `/api/v1/users/` is - the only problem is that [some urls took the form `/api/v1/users/abcd1234..../grants`](https://github.com/department-of-veterans-affairs/vets-api/pull/3784/files#diff-18132ff30b1bdbf5e33dc76464095e08R64).

I'm 🤷‍♂ and could go either way.